Dockerfile

```diff
@@ -1,93 +1,71 @@
-# Dockerfile to build an image able to compile flashlight for a variety of
-# platforms.
-#
-# > make docker
+# This docker machine is able to compile and sign Lantern for Linux and
+# Windows.
 
 FROM fedora:21
-MAINTAINER "José Carlos Nieto" <xiam@getlantern.org>
+MAINTAINER "The Lantern Team" <team@getlantern.org>
 
-ENV GO_VERSION go1.4.3
-ENV GOROOT_BOOTSTRAP /go1.4
-ENV GOROOT /go
-ENV GOPATH /
-
-ENV PATH $PATH:$GOROOT/bin
 ENV WORKDIR /lantern
+ENV SECRETS /secrets
 
-# Go binary for bootstrapping.
-ENV GO_PACKAGE_URL https://storage.googleapis.com/golang/go1.4.3.linux-amd64.tar.gz
+RUN mkdir -p $WORKDIR
+RUN mkdir -p $SECRETS
 
 # Updating system.
-RUN yum -y update && yum clean all
+RUN yum install -y deltarpm && yum update -y && yum clean packages
 
 # Requisites for building Go.
-RUN yum install -y git tar gzip curl hostname pcre-devel mercurial && yum clean all
+RUN yum install -y git tar gzip curl hostname && yum clean packages
 
 # Compilers and tools for CGO.
-RUN yum install -y gcc gcc-c++ libgcc.i686 gcc-c++.i686 pkg-config && yum clean all
-
-# Getting Go.
-RUN (curl -sSL $GO_PACKAGE_URL | tar -xvz -C /tmp) && \
-  mv /tmp/go $GOROOT_BOOTSTRAP
-
-# Getting Go source.
-RUN mkdir -p $GOROOT && \
-  git clone https://go.googlesource.com/go $GOROOT && \
-  cd $GOROOT && \
-  git checkout release-branch.go1.4
-
-# Patch for skipping a failing test in new docker versions. See https://github.com/getlantern/lantern/issues/2578
-RUN yum install -y patch && yum clean all
-RUN curl https://gist.githubusercontent.com/xiam/f50f6dd6085f9a07ccfd/raw/5e0f472221f9c1556fe34788ff01724b63980337/docker_golang | patch -p0
-
-# Bootstrapping Go.
-RUN cd $GOROOT/src && CGO_ENABLED=1 ./all.bash
+RUN yum install -y gcc gcc-c++ libgcc.i686 gcc-c++.i686 pkg-config && yum clean packages
 
 # Requisites for bootstrapping.
-RUN yum install -y glibc-devel glibc-static && yum clean all
-RUN yum install -y glibc-devel.i686 glib2-static.i686 glibc-2.20-8.fc21.i686 libgcc.i686 && yum clean all
+RUN yum install -y glibc-devel glibc-static && yum clean packages
+RUN yum install -y glibc-devel.i686 glib2-static.i686 glibc-2.20-8.fc21.i686 libgcc.i686 && yum clean packages
 
 # Requisites for ARM
 # ARM EABI toolchain must be grabbed from an contributor repository, such as:
 # https://copr.fedoraproject.org/coprs/lantw44/arm-linux-gnueabi-toolchain/
-RUN yum install -y yum-utils && yum-config-manager --add-repo=https://copr.fedoraproject.org/coprs/lantw44/arm-linux-gnueabi-toolchain/repo/fedora-21/lantw44-arm-linux-gnueabi-toolchain-fedora-21.repo && yum install -y arm-linux-gnueabi-gcc arm-linux-gnueabi-binutils arm-linux-gnueabi-glibc && yum clean all
+RUN yum install -y yum-utils && \
+  yum-config-manager --add-repo=https://copr.fedoraproject.org/coprs/lantw44/arm-linux-gnueabi-toolchain/repo/fedora-21/lantw44-arm-linux-gnueabi-toolchain-fedora-21.repo && \
```

Comment:
I encountered following warning:

> warning: /var/cache/yum/x86_64/21/lantw44-arm-linux-gnueabi-toolchain/packages/arm-linux-gnueabi-filesystem-2-1.fc21.noarch.rpm: Header V3 RSA/SHA1 Signature, key ID 1bf83516: NOKEY
> 
> Public key for arm-linux-gnueabi-filesystem-2-1.fc21.noarch.rpm is not installed

It is misleading when I check for failed build caused by other errors. I suggest we add below line before L30.
```
  rpm --import https://copr-be.cloud.fedoraproject.org/results/lantw44/arm-linux-gnueabi-toolchain/pubkey.gpg && \
```